binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -20,11 +55,32 @@ def iterate_new_and_changed_ebuilds(config):
         package = category_plus_package.split(os.sep)[-1]
 
         for ebuild_file in ebuild_files:
-            old_portdir_ebuild_file = os.path.join(config.old_portdir, category_plus_package,
-                                                   ebuild_file)
-            if os.path.exists(old_portdir_ebuild_file):
-                new_portdir_ebuild_file = os.path.join(root, ebuild_file)
-                if filecmp.cmp(old_portdir_ebuild_file, new_portdir_ebuild_file):
+            old_portdir_ebuild_filepath = os.path.join(config.old_portdir, category_plus_package,
+                                                       ebuild_file)
+            new_portdir_ebuild_filepath = os.path.join(root, ebuild_file)
+
+            # don't output 9999 ebuilds
+            if re.search(_filename_9999_pattern, ebuild_file) is not None:
+                continue
+
+            # don't output if files are identical
+            if os.path.exists(old_portdir_ebuild_filepath):
```

Comment:
Just an idea: I notice we run the same `os.path.exists(old_portdir_ebuild_filepath)` below later. I wonder if we should save this result into a variable, with the idea that file system access is way slower than RAM access (and I see no way for Python to cache results, internally). What do you think?

Reply:
I think the kernel does caching, yes, but we're coming from user space so you'd still have the price of the syscall going into the kernel and back. Nevermind, if it's not an issue let's not "fix" it, resisting premature optimisation can be hard, I guess doing the call twice is more readable and a good solution. Cool! :+1: